ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobSidecarsAvailabilityChecker.java

```diff
@@ -0,0 +1,131 @@
+      final Spec spec,
+      final SpecVersion specVersion,
+      final RecentChainData recentChainData,
+      final SignedBeaconBlock block,
+      final BlobSidecarsProvider blobSidecarsProvider) {
+    this.spec = spec;
+    this.specVersion = specVersion;
+    this.recentChainData = recentChainData;
+    this.block = block;
+    this.blobSidecarsProvider = blobSidecarsProvider;
+  }
+
+  @Override
+  public boolean initiateDataAvailabilityCheck() {
+    validationResult =
+        Optional.of(blobSidecarsProvider.getBlobSidecars(block).thenCompose(this::validate));
+    return true;
+  }
+
+  @Override
+  public SafeFuture<BlobSidecarsAndValidationResult> getAvailabilityCheckResult() {
+    return validationResult.orElse(NOT_REQUIRED_RESULT_FUTURE);
+  }
+
+  @Override
+  public SafeFuture<BlobSidecarsAndValidationResult> validate(
+      final List<BlobSidecar> blobSidecars) {
+    return SafeFuture.of(
+        () -> {
+          if (!blobSidecars.isEmpty()) {
+            return internalValidate(blobSidecars);
+          }
+
+          // When no blobs are available, it is ok to not have them (NOT_REQUIRED) if:
+
+          // 1. The block is not in the availability window
+          if (!isBlockInDataAvailabilityWindow()) {
+            return BlobSidecarsAndValidationResult.NOT_REQUIRED;
+          }
+          // 2. The number of kzg commitments in the block is 0
+          if (getNumberOfKzgCommitmentsInBlock() == 0) {
+            return BlobSidecarsAndValidationResult.NOT_REQUIRED;
+          }
+
+          return BlobSidecarsAndValidationResult.NOT_AVAILABLE;
+        });
+  }
+
+  private BlobSidecarsAndValidationResult internalValidate(final List<BlobSidecar> blobSidecars) {
+    final BeaconBlockBodyDeneb blockBody =
+        block
+            .getBeaconBlock()
+            .map(BeaconBlock::getBody)
+            .flatMap(BeaconBlockBody::toVersionDeneb)
+            .orElseThrow();
+    try {
+      if (!specVersion
+          .miscHelpers()
+          .isDataAvailable(
+              block.getSlot(),
+              block.getRoot(),
+              blockBody.getBlobKzgCommitments().stream()
+                  .map(SszKZGCommitment::getKZGCommitment)
+                  .collect(Collectors.toUnmodifiableList()),
+              blobSidecars)) {
+        return BlobSidecarsAndValidationResult.invalidResult(blobSidecars);
+      }
+    } catch (final Exception ex) {
+      return BlobSidecarsAndValidationResult.invalidResult(blobSidecars, ex);
+    }
+
+    return BlobSidecarsAndValidationResult.validResult(blobSidecars);
+  }
+
+  private boolean isBlockInDataAvailabilityWindow() {
+    return spec.isAvailabilityOfBlobSidecarsRequiredAtSlot(
+        recentChainData.getStore(), block.getSlot());
+  }
+
+  private int getNumberOfKzgCommitmentsInBlock() {
+    return block
+        .getMessage()
+        .getBody()
+        .toVersionDeneb()
+        .orElseThrow()
+        .getBlobKzgCommitments()
+        .size();
+  }
```

Comment:
We could define instead such method and use it twice
```java
  private int getKzgCommitments() {
    return block
        .getMessage()
        .getBody()
        .toVersionDeneb()
        .orElseThrow()
        .getBlobKzgCommitments();
  }
```

Reply:
you are right, thanks! this makes it easier